client/android/ZUMOAPPNAME/app/src/main/java/com/example/zumoappname/ToDoActivity.java

```diff
@@ -129,27 +154,41 @@ public void checkItem(final ToDoItem item) {
         // Set the item as completed and update it in the table
         item.setComplete(true);
 
-
-        new AsyncTask<Void, Void, Void>(){
+        AsyncTask<Void, Void, Void> task = new AsyncTask<Void, Void, Void>(){
             @Override
             protected Void doInBackground(Void... params) {
                 try {
-                    final ToDoItem entity = mToDoTable.update(item).get();
+
+                    checkItemInTable(item);
                     runOnUiThread(new Runnable() {
                         @Override
                         public void run() {
-                            if (entity.isComplete()) {
-                                mAdapter.remove(entity);
+                            if (item.isComplete()) {
+                                mAdapter.remove(item);
                             }
                         }
                     });
-                } catch (Exception e){
-                    createAndShowDialog(e, "Error");
+                } catch (final Exception e) {
+                    createAndShowDialogFromTask(e, "Error");
                 }
 
                 return null;
             }
-        }.execute();
+        };
+
+        runAsyncTask(task);
+
+    }
+
+    /**
+     * Mark an item as completed in the Mobile Service Table
+     *
+     * @param item
+     *            The item to mark
+     */
+    public void checkItemInTable(ToDoItem item) throws ExecutionException, InterruptedException {
+        mToDoTable.update(item).get();
+        //sync().get(); // offline sync
```

Comment:
This should not be present. Adding this would make application work only when the device is online. We should sync only when refresh button is pressed.